sklearn/tree/export.py

```diff
@@ -412,3 +413,107 @@ def recurse(tree, node_id, criterion, parent=None, depth=0):
+
+    height : height of resulting tree.
+
+    max_depth : int, optional (default=None)
+        The maximum depth of the representation. If None, the tree is fully
+        generated.
+
+    feature_names : list of strings, optional (default=None)
+        Names of each of the features.
+
+    class_names : list of strings, bool or None, optional (default=None)
+        Names of each of the target classes in ascending numerical order.
+        Only relevant for classification and not supported for multi-output.
+        If ``True``, shows a symbolic representation of the class name.
+
+    label : {'all', 'root', 'none'}, optional (default='all')
+        Whether to show informative labels for impurity, etc.
+        Options include 'all' to show at every node, 'root' to show only at
+        the top root node, or 'none' to not show at any node.
+
+    filled : bool, optional (default=False)
+        When set to ``True``, paint nodes to indicate majority class for
+        classification, extremity of values for regression, or purity of node
+        for multi-output.
+
+    leaves_parallel : bool, optional (default=False)
+        When set to ``True``, draw all leaf nodes at the bottom of the tree.
+
+    impurity : bool, optional (default=True)
+        When set to ``True``, show the impurity at each node.
+
+    node_ids : bool, optional (default=False)
+        When set to ``True``, show the ID number on each node.
+
+    proportion : bool, optional (default=False)
+        When set to ``True``, change the display of 'values' and/or 'samples'
+        to be proportions and percentages respectively.
+
+    rotate : bool, optional (default=False)
+        When set to ``True``, orient tree left to right rather than top-down.
+
+    rounded : bool, optional (default=False)
+        When set to ``True``, draw node boxes with rounded corners and use
+        Helvetica fonts instead of Times-Roman.
+
+    special_characters : bool, optional (default=False)
+        When set to ``False``, ignore special characters for PostScript
+        compatibility.
+
+    out : {'matplotlib', 'jupyter'}, optional (default='matplotlib')
+        Options include 'matplotlib' to plot using matplotlib library,
+        'jupyter' to plot to Jupyter Notebook environment.
+    """
+    try:
+        import graphviz
+    except ImportError:
+        raise ImportError(u'Plotting requires graphviz to be installed.')
+
+    in_memory_dot_file = six.StringIO()
+    export_graphviz(
+        decision_tree, out_file=in_memory_dot_file, max_depth=max_depth,
+        feature_names=feature_names, class_names=class_names, label=label,
+        filled=filled, leaves_parallel=leaves_parallel, impurity=impurity,
+        node_ids=node_ids, proportion=proportion, rotate=rotate,
+        rounded=rounded, special_characters=special_characters)
+    src = graphviz.Source(in_memory_dot_file.getvalue())
+    image = src.pipe(format='png')
+    if out == 'jupyter':
```

Comment:
Sorry for the nitpicking, but could you validate this out param at the start (before exporting) pl?